src/librbd/librbd.cc

```diff
@@ -2340,6 +2340,11 @@ extern "C" int rbd_trash_list(rados_ioctx_t p, rbd_trash_image_info_t *entries,
 
   vector<librbd::trash_image_info_t> cpp_entries;
   int r = librbd::trash_list(io_ctx, cpp_entries);
+  if (r == -ENOENT) {
```

Comment:
This only changes the behavior for the C API -- so it is leaving the C++ API inconsistent. It would probably be better to make a change like this within ```trash_list``` in internal.cc